engine/function/src/templates/Sort.ftl

```diff
@@ -30,7 +30,7 @@ public class Sort {
         if (values == null) {
             return null;
         }
-        if (values.size() == 0) {
+        if (values.isEmpty()) {
```

Comment:
Same comment about auditing `toArray()` as in `Basic.ftl`. The cases where we return an empty `toArray()` are fine. Sorting an un-copied `toArray()` is not. I see this issue on L38 (in `sortObj`) and L101 (in `sortDescendingObj`).

We copy aggressively in the primitive versions, at L162 and L220 respectively. This is safe, but might be an extra allocation and copy in some cases.